capirca/lib/paloaltofw.py

```diff
@@ -182,20 +183,27 @@ def __init__(self, from_zone, to_zone, terms, service_map):
                                                               to_zone)
       raise PaloAltoFWNameTooLongError(x)
 
-    self.options = {}
-    self.options["from_zone"] = [from_zone]
-    self.options["to_zone"] = [to_zone]
-    self.ModifyOptions(terms, service_map)
+    self.options = []
+    term = terms.term
+    while term is not None:
+      x, term = self.ModifyOptions(from_zone, to_zone,
+                                   term, service_map)
+      self.options.append(x)
 
-  def ModifyOptions(self, terms, service_map):
+  @staticmethod
+  def ModifyOptions(from_zone, to_zone, term, service_map):
```

Comment:
Nit: Should we rename this function now that it doesn't modify the options dict but rather just returns a new one?